xain/grpc/coordinator.py

```diff
@@ -103,112 +113,336 @@ def update_expires(self, participant_id):
             )
 
 
-def monitor_heartbeats(participants, terminate_event):
+class Coordinator:
+    """Class implementing the main Coordinator logic. It is implemented as a
+    state machine that reacts to received messages.
+
+    The states of the Coordinator are:
+        STANDBY: The coordinator is in standby mode, typically when waiting for
+        participants to connect. In this mode the only messages that the
+        coordinator can receive are :class:`~.coordinator_pb2.RendezvousRequest`
+        and :class:`~.coordinator_pb2.HeartbeatRequest`.
+
+        ROUND: A round is currently in progress. During a round the only
+        messages the coordinator can receive are
+        :class:`~.coordinator_pb2.StartTrainingRequest` and
+        :class:`~.coordinator_pb2.EndTrainingRequest`.
+
+        FINISHED: The training session has ended and participants should
+        disconnect from the coordinator.
+
+    States are exchanged during heartbeats so that both coordinators and
+    participants can react to each others state change.
+
+    The flow of the Coordinator:
+        1. The coordinator is started and waits for enough participants to join. `STANDBY`.
+        2. Once enough participants are connected the coordinator starts the rounds. `ROUND N`.
+        3. Repeat step 2. for the given number of rounds
+        4. The training session is over and the coordinator is ready to shutdown. `FINISHED`.
+
+    Note:
+        :class:`~.coordinator_pb2.RendezvousRequest` is always allowed
+        regardless of which state the coordinator is on.
+
+    Args:
+        num_rounds (:obj:`int`, optional): The number of rounds of the training
+            session. Defaults to 10.
+        required_participants(:obj:`int`, optional): The minimum number of
+            participants required to perform a round. Defaults to 10.
+        aggregator: (:class:`~.Aggregator`, optional): The type of aggregation
+            to perform at the end of each round. Defaults to
+            :class:`~.FederatedAveragingAgg`.
+        theta (:obj:`list` of :class:`~numpy.ndarray`, optional): The weights of
+            the global model. Defaults to [].
+        epochs (:obj:`int`, optional): TODO.  Defaults to 0.
```

Comment:
I can suggest a description (after I double-check it myself) if you like

Reply:
sure :+1:

Reply:
```suggestion
        epochs (:obj:`int`, optional): The number of rounds of training, internal to the Participant.  Defaults to 0.
```
(wrap the long line over where appropriate)